src/collection.js

```diff
@@ -3,6 +3,12 @@ import delay from 'xstream/extra/delay';
 import dropRepeats from 'xstream/extra/dropRepeats';
 import isolate from '@cycle/isolate';
 
+const noop = Function.prototype;
+
+function isVtree (x) {
+  return x && typeof x.sel === 'string';
```

Comment:
I wonder about the performance of this. 

Calling this for every event on every sink seems like a bit of a waste. I wonder if we can cache which sinks are `vtree`s and only call if we're unsure.

Probably premature optimization without benchmarks in place, but something to think about in future.